src/api/habits/adapters/json.ts

```diff
@@ -26,7 +30,75 @@ const jsonHabitServiceFactory = (): HabitService => {
         }))
       );
 
-  return { addHabit, getHabits };
+  const addEntry = (
+    goals: GoalWithHabitHistory[],
+    habitId: string,
+    date: Date = new Date(),
+    quantity: number = 1
+  ) => {
+    try {
```

Comment:
What does the try-catch do here? Since it's a known data transformation and not unknown input at this point, we should not expect an error

Reply:
I added the structure because I assumed that in the future when using a storage API, I'd be wanting to try-catch errors and resolve/reject the Promise accordingly.

Should I just resolve for now, for clarity?

Reply:
A try catch will not grab anything inside a Promise - a `.catch` will. Or unwrapping the promise with async/await, at which point try-catch works.

Also, you may wish to isolate the try-catch on an operation-basis - I understand now why you want to catch and it makes sense, but you may wish to make this clear and catch the specific operation. E.g.

```
const newData = await getNewDataSomehow()

try {
  await saveData(newData)
} catch (error) {
 console.log('saving error')
}

```

or in promise form,

```
getNewDataSomehow()
  .then(saveData)
  .catch(e => console.log('saving error')
```

Reply:
That makes sense. I mixed up the 2 methods. Thanks. Changed it now.